tidy3d/components/base.py

```diff
@@ -1014,7 +1014,7 @@ def insert_value(x, path: tuple[str, ...], sub_dict: dict):
             sub_element = current_dict[final_key]
             if isinstance(sub_element, DataArray):
                 current_dict[final_key] = sub_element.copy(deep=False, data=x)
-                if "AUTOGRAD" in sub_element.attrs:
+                if isbox(x):
```

Comment:
I think it was because `sub_element` might not have `.attrs[AUTOGRAD_KEY]` if somehow only the `.values` are being traced (and the `__init__` that does the `.values`/`.attrs` splitting never gets called). So `isbox(x)` will make sure that the `.attrs` will always be populated with tracers if `x` is being traced.

Reply:
I reverted this